### PR TITLE
[DOCS] Update re-ranking intro to remove confusion about stages

### PR DESCRIPTION
**Pull request by leemthompo**
h/t @2gavy for pointing out confusing verbiage here

### CODE REVIEW

File: docs/reference/reranking/index.asciidoc [hunk 1]
```diff
@@ -1,12 +1,12 @@
 [[re-ranking-overview]]
 = Re-ranking
 
-Many search systems are built on two-stage retrieval pipelines.
+Many search systems are built on multi-stage retrieval pipelines.
 
-The first stage uses cheap, fast algorithms to find a broad set of possible matches.
+Earlier stages uses cheap, fast algorithms to find a broad set of possible matches.
```
Comment (leemthompo) [inline docs/reference/reranking/index.asciidoc:6]:
```suggestion
Earlier stages use cheap, fast algorithms to find a broad set of possible matches.
```

File: docs/reference/reranking/index.asciidoc [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-The second stage uses a more powerful model, often machine learning-based, to reorder the documents.
-This second step is called re-ranking.
+Later stages uses a more powerful model, often machine learning-based, to reorder the documents.
```
Comment (leemthompo) [inline docs/reference/reranking/index.asciidoc:8]:
```suggestion
Later stages use more powerful models, often machine learning-based, to reorder the documents.
```

**Review by kderusso — APPROVED** [review]
Nice clarification, thanks!

**Comment by leemthompo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/114306)|
|✅|8.15|[](https://github.com/elastic/elasticsearch/pull/114308)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**